pkg/agent/apiserver/handlers/fqdncache/handler_test.go

```diff
@@ -26,48 +26,67 @@ import (
 	"github.com/stretchr/testify/require"
 	"go.uber.org/mock/gomock"
 
+	"antrea.io/antrea/pkg/agent/apis"
 	"antrea.io/antrea/pkg/agent/types"
 	queriertest "antrea.io/antrea/pkg/querier/testing"
 )
 
 func TestFqdnCacheQuery(t *testing.T) {
 	tests := []struct {
-		name             string
-		expectedStatus   int
-		expectedResponse []types.DnsCacheEntry
+		name                 string
+		expectedStatus       int
+		filteredCacheEntries []types.DnsCacheEntry
+		expectedResponse     []apis.FQDNCacheResponse
 	}{
 		{
 			name:           "FQDN cache exists - multiple addresses multiple domains",
 			expectedStatus: http.StatusOK,
-			expectedResponse: []types.DnsCacheEntry{
+			filteredCacheEntries: []types.DnsCacheEntry{
 				{
-					FqdnName:       "example.com",
-					IpAddress:      net.ParseIP("10.0.0.1"),
+					FQDNName:       "example.com",
+					IPAddress:      net.ParseIP("10.0.0.1"),
 					ExpirationTime: time.Date(2025, 12, 25, 15, 0, 0, 0, time.UTC),
 				},
 				{
-					FqdnName:       "foo.com",
-					IpAddress:      net.ParseIP("10.0.0.4"),
+					FQDNName:       "foo.com",
+					IPAddress:      net.ParseIP("10.0.0.4"),
 					ExpirationTime: time.Date(2025, 12, 25, 15, 0, 0, 0, time.UTC),
 				},
 				{
-					FqdnName:       "bar.com",
-					IpAddress:      net.ParseIP("10.0.0.5"),
+					FQDNName:       "bar.com",
+					IPAddress:      net.ParseIP("10.0.0.5"),
+					ExpirationTime: time.Date(2025, 12, 25, 15, 0, 0, 0, time.UTC),
+				},
+			},
+			expectedResponse: []apis.FQDNCacheResponse{
+				{
+					FQDNName:       "example.com",
+					IPAddress:      net.ParseIP("10.0.0.1").String(),
```

Comment:
```suggestion
					IPAddress:      "10.0.0.1",
```

same in other places below